src/Serialization/main.jl

```diff
@@ -133,6 +86,10 @@ end
 encodeType(::Type{<:Vector}) = "Vector"
 reverseTypeMap["Vector"] = Vector
 
+# special case for Matrix{T} specializations
+encodeType(::Type{<:Matrix}) = "Matrix"
+reverseTypeMap["Matrix"] = Matrix
+
```

Comment:
I like how you moved the custom `encodeType` for other contains to next where their save/load methods are, why not also do this for vector/matrix?

Reply:
good point, will do that